ui/src/app/styles/colors.ts

```diff
@@ -1,47 +1,28 @@
+import * as Color from 'color';
+
 export default {
-  purple: [
-    '#262262',
-    '#4356a7',
-    '#AB87B3'
-  ],
-  backgroundGrey: '#f1f2f2',
-  red: '#d0021b',
-  green: [
-    '#8bc990',
-    '#7AC79B',
-    '#7CC79B',
-    '#659F3D'
-  ],
-  blue: [
-    '#2691d0',
-    '#5aa6da',
-    '#6CACE4',
-    '#85bde4',
-    '#afd3ed',
-    '#d7eaf6',
-    '#eaf4fb',
-    '#2F2E7E',
-    '#302C71',
-    '#216FB4' // link blue
-  ],
-  gray: [
-    '#4a4a4a',
-    '#6e6e6e',
-    '#929292',
-    '#b7b7b7',
-    '#c3c3c3',
-    '#dbdbdb',
-    '#ededed',
-    '#F1F1F1'
-  ],
+  // Used for buttons and text
+  primary: '#262262',
+  // Used for tab navigation
+  secondary: '#6CACE4',
+  success: '#8BC990',
+  // Used for tables
+  light: '#E9ECEF',
+  highlight: '#F8C954',
+  warning: '#F7981C',
+  danger: '#DB3214',
+  dark: '#4A4A4A',
+  // Text links and Icon links
+  accent: '#216FB4',
+  resourceCardHighlights: {
+    conceptSet: '#A27BD7',
+    cohort: '#F8C954',
+    notebook: '#8BC990',
+    dataSet: '#6CACE4'
+  },
   white: '#fff',
-  black: [
-    '#000',
-    '#333333'
-  ],
-  yellow: [
-    '#F8C954',
-    '#F7981C',
-    '#F68D76'
-  ]
+};
+
+export const colorWithLightness = (color: string, intensity: number) => {
+  return Color(color).mix(Color('white'), 1 - intensity).toString();
```

Comment:
The mixing is an interesting approach, but would color.lighten (or color.whiten) be simpler and do something similar if not identical?

Reply:
color.lighten actually affected the rgb values, because I think I would need to mess with hue/saturation as well to get it to work. This is the approach Terra uses, and also lets us use higher numbers to darken, so I wanted to go with their approach.

Reply:
Talking to Isaac later. I am tempted to just make it intensity though, if what Brett said is correct.
(`.mix` gives us a range of -1 to 1, where -1 is black, 1 is white, and 0 is the base color)